### DISCUSSION TITLE
ParaView MFEM Reader

### DISCUSSION THREAD

**Discussion by benzwick** [mfem/mfem#2234] [Show and tell]
I have created a ParaView plugin for reading MFEM mesh files:

https://github.com/benzwick/paraview-mfem-reader

I will leave the link here in case anyone else finds it useful. So far only straight meshes (v1.0) are supported but it should not be too difficult to add more general curved meshes in the future.

If anyone has any ideas for how to load MFEM grid functions at the same time as loading the mesh that would be great too but I think that would require quite a bit of extra work.

**Comment by benzwick** [comment]
I made a few improvements:

- Add support for `mfem_root` (VisItDataCollection) files.
- Use PyMFEM to load the mesh and grid functions.

This makes it possible to open MFEM data collections inside ParaView without using VTK files.

**Comment by tzanio** [comment]
ping: @pazner, @jandrej, @bslazarov
